addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -200,6 +221,14 @@ def action_approve(self):
         if any(holiday.state != 'confirm' for holiday in self):
             raise UserError(_('Leave request must be confirmed ("To Approve") in order to approve it.'))
 
+        for holiday in self:
+            validation_type = holiday.holiday_status_id.validation_type
+            department = holiday.employee_id.department_id
+            if (holiday.holiday_status_id.double_validation or validation_type == 'manager') and department.manager_id != current_employee:
+                raise ValidationError('You must be {} manager to approve this leave'.format(department.name))
```

Comment:
`ValidationError` should be raised only in constraint method. Use `UserError` here.